site/_js/misc/render-event-card.js

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {html} = require('common-tags');
+const {formatDateShort} = require('../../_data/lib/date');
+
+/**
+ * Renders an event card.
+ *
+ * @param {object} event
+ * @param {object} icons
+ * @returns string
+ */
+function RenderEventCard(event, icons) {
+  const buttonLabel = event.isPastEvent ? 'See details' : "See who's joining";
+
+  return html`
+    <enhanced-event-card>
```

Comment:
This is used on backend code, right? in _js/ folder, only frontend js files should live.

If you want to use this as aboth backend lib and frontend lib, I worry that:
1. we will run into issues using require() modules in frontend code
2. you use a LOT of js, with additional imports, to achieve fairly straightforward declarative html output.

How about rendering the events on backend, and sending down html partials instead of json? or embed output html as json field? 

Or alternatively, making it a web component, and always render on the client side?